docs/miscellaneous.rst

```diff
@@ -56,6 +56,20 @@ So for the following contract snippet::
 
 The position of ``data[4][9].b`` is at ``keccak256(uint256(9) . keccak256(uint256(4) . uint256(1))) + 1``.
 
+****************
+Layout in Memory
+****************
+
+EVM memory is linear. Reading is limited to 256-bit width, but both 8-bit and 256-bit writes are allowed. Memory size is 0 at the beginning of execution. Memory is expanded by accessing (either reading or writing) a previously untouched memory offset. At the time of expansion, the cost in gas must be paid.
+
+Solidity reserves two 256-bit slots:
+-  0 - 32: scratch space for hashing methods
```

Comment:
I thought the scratch place has two words.  It has 64 bytes (that useful for array index calculation).